src/main/java/org/sagebionetworks/web/client/SynapseJavascriptClient.java

```diff
@@ -947,6 +947,25 @@ public void getRestrictionInformation(
     );
   }
 
+  public FluentFuture<RestrictionInformationResponse> getRestrictionInformation(
```

Comment:
Because the download menu was incorporated into the EntityActionMenu, these two new methods are now called from EntityActionController, which is using futures for asynchronous fetching.